kendzi3d-josm-core/src/main/java/kendzi/josm/kendzi3d/util/ModelUtil.java

```diff
@@ -64,32 +64,29 @@ public static Double parseHeight(String pHeightStr, Double pDefault) {
             return pDefault;
         }
 
-        String heightStr = pHeightStr.trim().toLowerCase();
-        heightStr = heightStr.replaceAll(",", ".");
+        pHeightStr = pHeightStr.trim().toLowerCase().replaceAll(",", ".");
 
         try {
-            if (heightStr.endsWith("m")) {
-                heightStr = heightStr.substring(0, heightStr.length() - 1);
-                return new Double(" " + heightStr + " ");
-
-            } else if (heightStr.endsWith("ft")) {
-
-                // 1 foot = 0.3048 meters
-                heightStr = heightStr.substring(0, heightStr.length() - 1);
-                return 0.3048d * new Double(" " + heightStr + " ");
-
-            } else if (heightStr.endsWith("feet")) {
-
-                // 1 foot = 0.3048 meters
-                heightStr = heightStr.substring(0, heightStr.length() - 4);
-                return 0.3048d * new Double(" " + heightStr + " ");
+            // default we take in [m]
+            double mult = 1d;
+            int strlen = pHeightStr.length();
+
+            switch (pHeightStr.replaceAll("[^a-z]", "")) {
```

Comment:
Looks nice, but it will be killed by auto formatter. I could promise i write about using auto formatter in past, but looks it get lost by time. I will export settings and try to document. But don't worry, it is only detail.

Reply:
replaceAll()  is  just applied to retain potential units in the tag value
(e.g.  "4.75 m"  -> "m", or "8 feet" -> "feet",  "1.3432 km" -> "km")
--> this is just a more concise way of doing endsWith() checks
--> replaceAll() will not change pHeightStr in any way, it will produce new value that is only needed for switch

Or do you mean the input to parseHeight() is normalized before it is called?  The function checked unit suffixes before my changes with endsWith(), so it should work fine.  I will double check with debug print, however and report back.

Reply:
Sorry I was not clear, I'm referring only to source code formatting. You format your switch manually, and I get used to auto-fromatters after save action. Anyway your code looks correct.

Reply:
used eclipse auto-format in the latest commit for that file..

I don't know if you use eclipse or other formatter, but it will definitely help if you document (or commit to git) the settings you use to format the source code.

Imho source code formatting is a negligible detail as long as it is human-readable and no automatic code-obfuscation was used.  If you've seen obfuscated javascript source-code with no line breaks and almost no white space - that's a non-negligible counter example.